### PR TITLE
Remove .NET Core 3.1 + .NET 5 and add .NET 7

### PR DESCRIPTION
**Pull request by amoerie**
Removes .NET Core 3.1 and .NET 5 from our test target frameworks and adds .NET 7
According to https://dotnet.microsoft.com/en-us/platform/support/policy/dotnet-core these runtimes are now out of support, so I think it makes sense for Fellow Oak DICOM to follow that same pattern.

#### Checklist
- [x] The pull request branch is in sync with latest commit on the *fo-dicom/development* branch
- [x] I have updated API documentation
- [x] I have included unit tests
- [x] I have updated the change log
- [x] I am listed in the CONTRIBUTORS file

#### Changes proposed in this pull request:
- Remove .NET Core 3.1 and .NET 5.0 from TargetFrameworks
- Add .NET 7.0
- Update Github build.yml 
- Add chapter in README on supported runtimes

Note: this is not a breaking change. 
Fellow Oak DICOM still "supports" .NET Standard 2.0 and will continue to work on .NET Core 5.0 or .NET Core 3.1
The difference is that our unit tests will no longer run against these runtimes, so these might break in the future.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,6 +21,16 @@ This library is licensed under the [Microsoft Public License (MS-PL)](http://ope
...
 
+### Supported Runtimes
+
+Fellow Oak DICOM officially supports the following runtimes:
+
+* .NET Core 7.0
+* .NET Core 6.0
+* .NET Framework 4.6.2
+
+Other runtimes that implement .NET Standard 2.0 may work, but are not tested by Fellow Oak DICOM maintainers.
```
Comment (mrbean-bremen) [inline README.md]: Maybe better something like "tested by the CI environment", and also mention that the tests currently only run under Windows.

File: Tests/FO-DICOM.Tests/Helpers/SerializationExtensions.cs [hunk 2]
```diff
@@ -26,31 +26,5 @@ public static T GetDataContractSerializerDeserializedObject<T>(this T tag)
             deserializedObject = (T)serializer.ReadObject(stream);
             return deserializedObject;
         }
-
-        public static T GetBinaryFormatterDeserializedObject<T>(this T input)
```
Comment (mrbean-bremen) [inline Tests/FO-DICOM.Tests/Helpers/SerializationExtensions.cs:30]: I guess this change is not related to the PR?

File: Tests/FO-DICOM.Tests/Imaging/Mathematics/Point2Tests.cs [hunk 3]
```diff
@@ -19,12 +18,5 @@ public void Serialization_RegularPoint_CanBeDeserialized()
             var actual = expected.GetDataContractSerializerDeserializedObject();
             Assert.Equal(expected, actual);
         }
-
-        [Fact]
-        public void Serialization_BinaryFormatter_Throws()
```
Comment (mrbean-bremen) [inline Tests/FO-DICOM.Tests/Imaging/Mathematics/Point2Tests.cs:24]: Same here

File: Tests/FO-DICOM.Tests/Helpers/SerializationExtensions.cs (hunk 2 above)
Comment (amoerie) [reply Tests/FO-DICOM.Tests/Helpers/SerializationExtensions.cs:30]: BinaryFormatter is deprecated in .NET 7 and our usage is only test related, so it seemed natural to just remove the offending test.

File: Tests/FO-DICOM.Tests/Imaging/Mathematics/Point2Tests.cs (hunk 3 above)
Comment (amoerie) [reply Tests/FO-DICOM.Tests/Imaging/Mathematics/Point2Tests.cs:24]: BinaryFormatter is deprecated in .NET 7 and our usage is only test related, so it seemed natural to just remove the offending test.

Comment (mrbean-bremen) [reply Tests/FO-DICOM.Tests/Imaging/Mathematics/Point2Tests.cs:24]: ah ok, thanks!

**Review by mrbean-bremen — APPROVED** [review]

**Review by gofal — APPROVED** [review]
Thanks!

File: README.md (hunk 1 above)
Comment (amoerie) [reply README.md]: Good suggestions, done
